jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/OdeTimJsonTopology.java

```diff
@@ -64,19 +72,42 @@ public boolean isRunning() {
    */
   public Topology buildTopology(String topic) {
     StreamsBuilder builder = new StreamsBuilder();
-    builder.table(topic,
-        Materialized.<String, String>as(Stores.inMemoryKeyValueStore("timjson-store")));
+
+    // Create a windowed store with a retention period of 1 hour
+    KStream<String, String> timStream = builder.stream(topic);
+
+    timStream.groupByKey()
+            .windowedBy(TimeWindows.ofSizeWithNoGrace(Duration.ofHours(1))) // 1 hour window
+            .reduce(
+                    (aggValue, newValue) -> newValue, // only keep latest value
+                    Materialized.<String, String, WindowStore<Bytes, byte[]>>as("timjson-windowed-store")
+                            .withRetention(Duration.ofHours(1))
+                            .withKeySerde(Serdes.String())
+                            .withValueSerde(Serdes.String()));
+
     return builder.build();
   }
 
   /**
-   * Query the K-Table by a specified UUID.
+   * Query the windowed store by a specified UUID.
    *
    * @param uuid The specified UUID to query for.
    **/
   public String query(String uuid) {
-    return (String) streams.store(
-            StoreQueryParameters.fromNameAndType("timjson-store", QueryableStoreTypes.keyValueStore()))
-        .get(uuid);
+    ReadOnlyWindowStore<String, String> windowStore =
+        streams.store(StoreQueryParameters.fromNameAndType("timjson-windowed-store", QueryableStoreTypes.windowStore()));
+
+    Instant now = Instant.now();
+    Instant start = now.minus(Duration.ofHours(1));
+    String value;
+    
+    try (WindowStoreIterator<String> iterator = windowStore.fetch(uuid, start, now)) {
+      value = null;
+      while (iterator.hasNext()) {
+        value = String.valueOf(iterator.next().value);
+      }
+    }
+    
+    return value;
```

Comment:
suggestion: since we know that we should only ever have one value per uuid in the table, we can use the other `fetch` available that only returns the value (if found) or `null` if not found. That way we don't need to iterate through a list when we know there's only one value.

```suggestion
    Instant start = Instant.now().minus(Duration.ofHours(1));
    
    return windowStore.fetch(uuid, start.toEpochMilli());
```